### PR TITLE
nx-table updates - RSC-177

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-177

### CODE REVIEW

**Review by Blacktiger — APPROVED** [review]
I take it `metaInfo` is an internal prop and doesn't need any documentation? What's the purpose of that prop?

**Comment by rpokorny** [comment]
> I take it metaInfo is an internal prop and doesn't need any documentation? What's the purpose of that prop?

That's how I intended it but I was thinking last night maybe it actually should be documented.  It just adds the `.nx-cell--meta-info` class, which is now the class that should be used for "special" cells that stretch all the way across.  The three current use cases being the empty, loading, and error states.  But I realized consumers might have their own additional scenarios requiring a meta-row like that, so I think I'll add docs for it.

**Comment by glenhunter** [comment]
I'd like to see a table with fewer columns but longer text similar to what Derrek has in Zeplin, just to make sure that the text wraps and aligns correctly

**Comment by glenhunter** [comment]
Derrek also shows wrapping and truncation in table cells so it would be nice to see examples of both

**Comment by rpokorny** [comment]
> That's how I intended it but I was thinking last night maybe it actually should be documented. It just adds the .nx-cell--meta-info class, which is now the class that should be used for "special" cells that stretch all the way across. The three current use cases being the empty, loading, and error states. But I realized consumers might have their own additional scenarios requiring a meta-row like that, so I think I'll add docs for it.

3e79296

File: lib/src/base-styles/_nx-alert.scss [hunk 1]
```diff
@@ -19,6 +19,7 @@
   display: flex;
   margin: 8px 0;
   padding: $nx-spacing-xs $nx-spacing-md $nx-spacing-xs $nx-spacing-md;
+  text-align: initial;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-alert.scss:22]: Needed for when a generic `NxAlert` is used within an `.nx-cell--meta-info`

**Review by Blacktiger — APPROVED** [review]

**Comment by rpokorny** [comment]
> Derrek also shows wrapping and truncation in table cells so it would be nice to see examples of both

e613c06 and also note a5c06d4

File: lib/src/base-styles/_nx-tables.scss [hunk 2]
```diff
@@ -10,141 +10,145 @@
...
 }
 
-// Used when table contents are empty
-.nx-cell--empty {
-  @include regular();
-
-  color: $nx-text-color;
+// Used for table cells containiner meta-information such as the fact that the table is empty, or
+// that the data is loading. These cells are expected to take up the entire row using colspan attributes
+.nx-cell--meta-info {
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-tables.scss:160]: Zeplin shows the error, empty, and alert cells as having 16px of top spacing

Comment (rpokorny) [reply lib/src/base-styles/_nx-tables.scss]: > Zeplin shows the error, empty, and alert cells as having 16px of top spacing

78c2c7f0

**Comment by rpokorny** [comment]
Just noticed scrolling table styles have broken, looking into it.

**Comment by glenhunter** [comment]
@dturner32 I think that we should have a `selected` state for table rows.

**Comment by glenhunter** [comment]
@dturner32 it seems od to me that in footers and lists the error alerts are the full width of the container but in tables they are constrained in width

File: lib/src/base-styles/_nx-tables.scss [hunk 3]
```diff
@@ -10,141 +10,173 @@
...
-.nx-cell--label {
-  @include semi-bold();
-
-  color: $nx-text-color;
+  white-space: nowrap;
 }
 
 // columns whose content is numerical
 .nx-cell--num {
   text-align: center;
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-tables.scss:170]: The Zeplin files show numbers left aligned, but I thought there had been a conversation about numbers being centred. @dturner32 ?

**Comment by rpokorny** [comment]
> Ah, I see that Zeplin shows them as constrained but the examples here are full width - so which is correct?

@dturner32 and I discussed this while I was implementing.  Apparently, all `nx-alert`s are supposed to have a 950px max width.  That was never implemented and we need to put in a bug about it.  That said, I believe the examples here fall under that max width anyway.

**Comment by rpokorny** [comment]
> sp: "container"

0e463fa2

File: lib/src/base-styles/_nx-tables.scss (hunk 3 above)
Comment (dturner32) [reply lib/src/base-styles/_nx-tables.scss:177]: > The Zeplin files show numbers left aligned, but I thought there had been a conversation about numbers being centred. @dturner32 ?

yes they should be centered, I just forgot to update the mocks in zeplin. I will fix that

**Comment by dturner32** [comment]
> @dturner32 I think that we should have a `selected` state for table rows.

I can add that in

**Comment by dturner32** [comment]
> > @dturner32 I think that we should have a `selected` state for table rows.
> 
> I can add that in

just added selected states for table rows and lists to zeplin

File: lib/src/base-styles/_nx-tables.scss (hunk 3 above)
Comment (dturner32) [reply lib/src/base-styles/_nx-tables.scss:177]: > > The Zeplin files show numbers left aligned, but I thought there had been a conversation about numbers being centred. @dturner32 ?
> 
> yes they should be centered, I just forgot to update the mocks in zeplin. I will fix that

just updated zeplin

File: lib/src/base-styles/_nx-tables.scss [hunk 4]
```diff
@@ -10,141 +10,180 @@
...
-
-.nx-cell--icon {
-  color: $nx-text-color;
-  padding: 0 $nx-spacing-sm;
-  text-align: right;
-  width: 14px;
-  max-width: 32px;
-  min-width: 14px;
+  // seems there's still no cleaner way to get a cell to shrink to fit its contents
+  width: 1px;
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-tables.scss:186]: If you add multiple icons they appear vertically. This doesn't seem optimal. Are you thinking that in that case each icon would have a separate cell?
Screen Shot 2020-09-29 at 4 38 27 PM

Comment (glenhunter) [reply lib/src/base-styles/_nx-tables.scss:186]: Or better yet could we have it adapt to multiple icons?

Comment (rpokorny) [reply lib/src/base-styles/_nx-tables.scss]: > Or better yet could we have it adapt to multiple icons?

I've separated out the styles for the chevron column vs general icon columns, allowing general icon columns to now contain multiple icons. 9cd195e0

**Review by glenhunter — APPROVED** [review]

**Pull request merged**